src/openms/include/OpenMS/ANALYSIS/OPENSWATH/PeakPickerMobilogram.h

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) 2002-present, The OpenMS Team -- EKU Tuebingen, ETH Zurich, and FU Berlin
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Justin Sing $
+// $Authors: Justin Sing $
+// --------------------------------------------------------------------------
+
+#pragma once
+
+#include <OpenMS/DATASTRUCTURES/DefaultParamHandler.h>
+#include <OpenMS/CONCEPT/ProgressLogger.h>
+#include <OpenMS/KERNEL/MSSpectrum.h>
+#include <OpenMS/KERNEL/MSChromatogram.h>
+#include <OpenMS/KERNEL/ChromatogramPeak.h>
+#include <OpenMS/KERNEL/Mobilogram.h>
+#include <OpenMS/KERNEL/MobilityPeak1D.h>
+
+#include <OpenMS/PROCESSING/NOISEESTIMATION/SignalToNoiseEstimatorMedian.h>
+#include <OpenMS/PROCESSING/SMOOTHING/SavitzkyGolayFilter.h>
+#include <OpenMS/PROCESSING/SMOOTHING/GaussFilter.h>
+
+#include <OpenMS/PROCESSING/CENTROIDING/PeakPickerHiRes.h>
+#include <vector>
+
+namespace OpenMS
+{
+    /**
+    @brief The PeakPickerMobilogram finds peaks a single mobilogram.
+
+    @htmlinclude OpenMS_PeakPickerMobilogram.parameters
+
+    It uses the PeakPickerHiRes internally to find interesting seed candidates.
+    These candidates are then expanded and a right/left border of the peak is
+    searched.
+    Additionally, overlapping peaks can be removed.
+
+    */
+
+    class OPENMS_DLLAPI PeakPickerMobilogram :
+            public DefaultParamHandler
+    {
+    public:
+        //@{
+        /// Constructor
+        PeakPickerMobilogram();
+
+        /// Destructor
+        ~PeakPickerMobilogram() override {}
+        //@}
+
+        /// indices into FloatDataArrays of resulting picked mobilogram
+        enum FLOATINDICES { IDX_FWHM = 0, IDX_ABUNDANCE = 1, IDX_LEFTBORDER = 2, IDX_RIGHTBORDER = 3, SIZE_OF_FLOATINDICES };
+
+        /// Struct to hold peak positions
+        struct PeakPositions {
+            size_t left;    // Left position of the peak
+            size_t apex;    // Apex (highest point) position of the peak
+            size_t right;   // Right position of the peak
+        };
+
+        /**
+            @brief Finds peaks in a single mobilogram and annotates left/right borders
+
+            It uses a modified algorithm of the PeakPickerHiRes
+
+            This function will return a picked mobilogram
+        */
+        void pickMobilogram(const Mobilogram& mobilogram, Mobilogram& picked_mobilogram);
+
+        /**
+            @brief Finds peaks in a single mobilogram and annotates left/right borders
+
+            It uses a modified algorithm of the PeakPickerHiRes
+
+            This function will return a picked mobilogram and a smoothed mobilogram
+        */
+        void pickMobilogram(Mobilogram mobilogram, Mobilogram& picked_mobilogram, Mobilogram& smoothed_mobilogram);
```

Comment:
Is it possible to break this up into a function
smoothMobilogram and pickMobilogram ?
Looks a bit like this method could be doing a bit too much...

Reply:
I think the smoothing is already somewhat separated? I could move the section below out into a separate function? This is the only part for the smoothing

```
// Smooth the mobilogram
smoothed_mobilogram = mobilogram;
if (!use_gauss_)
{
    sgolay_.filter(smoothed_mobilogram);
}
else
{
    gauss_.filter(smoothed_mobilogram);
}
```